packages/product/src/models/product-variant-option.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Entity,
+  Filter,
+  Index,
+  ManyToOne,
+  OnInit,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import { ProductOptionValue, ProductVariant } from "./index"
+
+const variantOptionValueIndexName = "IDX_variant_option_option_value_unique"
+const variantOptionValueIndexStatement = createPsqlIndexStatementHelper({
+  name: variantOptionValueIndexName,
+  tableName: "product_variant_option",
+  columns: ["variant_id", "option_value_id"],
+  unique: true,
+  where: "deleted_at IS NULL",
+})
+
+variantOptionValueIndexStatement.MikroORMIndex()
+@Entity({ tableName: "product_variant_option" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+class ProductVariantOption {
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @Property({ columnType: "text", nullable: true })
+  option_value_id!: string
+
+  @ManyToOne(() => ProductOptionValue, {
+    fieldName: "option_value_id",
+    nullable: true,
+  })
+  option_value!: ProductOptionValue
+
+  @Property({ columnType: "text", nullable: true })
+  variant_id!: string
+
+  @ManyToOne(() => ProductVariant, {
+    fieldName: "variant_id",
+    nullable: true,
```

Comment:
q: should these (variant_id and option_value_id) be nullable? For a pivot entity, these should be required right?

Reply:
@riqwan There is annoying issue where you might only pass one or the other and MikroORM will complain about it. I think there is a quite a bit of work left to do the modeling properly, for now I just do whatever works to get the module passing tests :)